supervisor/addons/options.py

```diff
@@ -241,8 +241,15 @@ def _check_missing_options(
         """Check if all options are exists."""
         missing = set(origin) - set(exists)
         for miss_opt in missing:
-            if isinstance(origin[miss_opt], str) and origin[miss_opt].endswith("?"):
+            miss_schema = origin[miss_opt]
+
+            # If its a list then value in list decides if its optional like ["str?"]
+            if isinstance(miss_schema, list) and len(miss_schema) > 0:
+                miss_schema = miss_schema[0]
```

Comment:
We should instead check each item in the list if they are optional or not.

Now it doesn't work with lists containing dicts with items.

Reply:
So like if its a list of objects and each key in the object is optional then make the entire object optional? That makes sense although objects might also have other object/list fields so its a bit more complicated. You'd kind of have to recursively look through each object field you found to answer the question of whether its optional or not.

I'll think about it. I feel like it should be a separate PR regardless right? It's a separate gap and I don't think its trivial because of the nested fields problem.

Reply:
Also just to clarify, the previous code here was this:

```python
 def _check_missing_options(
    self, origin: Dict[Any, Any], exists: Dict[Any, Any], root: str
) -> None:
    """Check if all options are exists."""
    missing = set(origin) - set(exists)
    for miss_opt in missing:
        miss_schema = origin[miss_opt]

        if isinstance(miss_schema, str) and miss_schema.endswith("?"):
            continue

        raise vol.Invalid(
            f"Missing option '{miss_opt}' in {root} in {self._name} ({self._slug})"
        ) from None
```

As you can see, it only had handling for fields which were of type string in the schema. If the schema (`origin`) said a field wasn't type string then it simply assumed it was required. I added the capability to have a config which is an array of strings be optional but there's no change to how configs of type `dict` or `list(dict)` are handled, they will always be considered required.

Note that this doesn't mean we are no longer validating dictionaries in arrays if that's your concern. This method is only responsible for looking at the provided dictionary and ensuring all required fields have a value. Other methods go through each dictionary in each list and field and make sure they meet validation requirements.

Reply:
My concern was this fixes one of the possible problems. Would have been nice if we could do the same for dicts in the list.